compiler/optimizer/ExpressionsSimplification.cpp

```diff
@@ -812,6 +814,12 @@ TR_ExpressionsSimplification::findLoopInfo(TR_RegionStructure* region)
          equals = true;
       case TR::ificmple:
       case TR::ificmpge:
+         {
+         TR::TreeTop *branchDestTT = lastTreeInExitBlock->getBranchDestination();
+         TR::Block *branchDest = branchDestTT->getNode()->getBlock();
+         if (branchDest->getNumber() != exitTarget)
```

Comment:
Would a "branch dest == loop header" test be safer than "branch dest != exit target" test here ? 
The loop in question probably does'nt allow more than one exit currently but is it better to future proof it via the other test ?

Reply:
The in-loop target is not necessarily the loop header. For example, consider this loop:
```
int a = ...;
int i = 0;
forever {
    foo();
    sum += a;
    if (++i >= 10) break;
    bar();
}
```
The conditional could be reversed with the loop laid out like this:
```
    int a = ...;
    int i = 0;
    goto header;
tail:
    bar();
header:
    foo();
    sum += a;
    if (++i < 10) goto tail;
exit:
```
Furthermore, if there is another exit, then there is no way to accurately predict the number of loop iterations. The `lastTreeInExitBlock` provides a bound, but any smaller number of iterations is possible. Though I suppose that in the future the computation could be moved into loop exits instead, where it would be possible to subtract to get the number of iterations that actually happened.